src/adr/0040-quality-levels-definitions.md

```diff
@@ -0,0 +1,62 @@
+# 40. Quality levels definition
+
+Date: 2020-06-21
+
+## Status
+
+Accepted (lazy consensus)
+
+## Context
+
+We hereby define as an artifact compiled artifact that external people consumes. This includes:
+
+ - libraries
+ - Mail servers
+ - Extensions for James Mail Servers
+ - Command line tools
+
+We designate as a feature an optional, opt-in behaviour of a James server that can be configured by 
+user willing to rely on it.
+
+James as a project delivers several artifacts, and features. In order for project users to better
```

Comment:
These seem to me like important and fundamental enough concepts to either expand the scope of this ADR, or to have a separate ADR that clearly defines (as a base concept for other ADRs like this one) the concepts of "artifact" and "feature".

Reply:
I think the context part does it well enough. If it is not clear enough, please provide alternative wordings.

Reply:
That is not quite what I meant. I was not commenting on the words, I was commenting on the scope. This ADR is about "quality". However, the terms "artifacts" and "features" seem to be so fundamental that they should not be hidden away in a document that describes something different.

So we should either:

1. Expand the scope of this document (i.e. it defines "Artifacts", "Features", and "Quality Levels" and not just "Quality Levels", or
2. Have a separate ADR that defines "Artifacts" and "Features".

Then this can be referred to from other discussions which will surely use these fundamental concepts.

Just something to consider, maybe.

Reply:
I have a different approach here.

I started another document to explain which server we ship as part of the James project, then I hope that with @ieugen we come up with a list of libraries we deliver. Each one of them will get their own ADR, giving a de-facto definition (and exhaustive list of artifacts).

Regarding features, refining extension mechanisms and planning the work for their extraction would do the same.